dnf/base.py

```diff
@@ -1413,22 +1413,32 @@ def cond_check(pkg):
                 pkg.requires in (pkg.name for pkg in self._goal._installs) or \
                 pkg.requires in [pkg.name for pkg in trans.install]
 
-        def trans_upgrade(query, remove_query):
+        def trans_upgrade(query, remove_query, pkg):
             sltr = dnf.selector.Selector(self.sack)
             sltr.set(pkg=query)
             self._goal.upgrade(select=sltr)
             return remove_query
 
-        def trans_install(query, remove_query):
+        def trans_install(query, remove_query, pkg):
             if self.conf.multilib_policy == "all":
-                self._install_multiarch(query, strict=False)
+                if not pkg.requires:
+                    self._install_multiarch(query, strict=False)
+                else:
```

Comment:
Change to `elif cond_check(pkg):`, please. Also rename `cond_check()` or comment the `cond_check` function, so the user knows what the function does.
￼